web/crux/src/app/deploy/pipes/deploy.secrets.pipe.ts

```diff
@@ -0,0 +1,88 @@
+import { Injectable, PipeTransform, Logger } from '@nestjs/common'
+import PrismaService from 'src/services/prisma.service'
+import { IdRequest } from 'src/grpc/protobuf/proto/crux'
+import AgentService from 'src/app/agent/agent.service'
+import { UniqueSecretKeyValue } from 'src/shared/model'
+
+@Injectable()
+export default class DeploySecretsValidationPipe implements PipeTransform {
+  private readonly logger = new Logger(DeploySecretsValidationPipe.name)
+
+  constructor(private prisma: PrismaService, private agentService: AgentService) {}
+
+  async transform(value: IdRequest) {
+    const deployment = await this.prisma.deployment.findUniqueOrThrow({
+      where: {
+        id: value.id,
+      },
+      select: {
+        nodeId: true,
+      },
+    })
+
+    const publicKey = this.agentService.getById(deployment.nodeId)?.publicKey
+
+    if (!publicKey) {
+      return value
+    }
+
+    const deploymentConfig = await this.prisma.deployment.findUniqueOrThrow({
```

Comment:
Why are you making the same transaction twice instead of adding `{select.nodeId:true}` to this one?

Reply:
I'm currently refactoring this after talking with @m8vago. I've already removed this file :)

That's why I moved the PR back to draft.

Reply:
Ahh, okay. I didn't get the information. (didn't refreshed the page 🤡 )